agent/engine/task_manager.go

```diff
@@ -802,6 +947,24 @@ func (mtask *managedTask) containerNextState(container *api.Container) *containe
 	}
 }
 
+func (mtask *managedTask) resourceNextState(resource taskresource.TaskResource) *resourceTransition {
+	var nextState taskresource.ResourceStatus
```

Comment:
you really don't need this var. It's adding 3 more additional lines of code.

Reply:
We are reusing `nextState` var in `statusString` field too, hence left it there.

Reply:
Hmm, alright. You can still reduce 1 line by doing `nextState := resource.TerminalStatus()` in both of those places :)